programs/mmm/src/instructions/cnft/sol_cnft_fulfill_buy.rs

```diff
@@ -0,0 +1,418 @@
+        }
+        transfer_compressed_nft(
+            &ctx.accounts.tree_authority.to_account_info(),
+            &pool.to_account_info(),
+            &ctx.accounts.payer.to_account_info(),
+            &ctx.accounts.owner.to_account_info(),
+            &ctx.accounts.merkle_tree,
+            &ctx.accounts.log_wrapper,
+            &ctx.accounts.compression_program,
+            &ctx.accounts.system_program, // Pass as Program<System> without calling to_account_info()
+            proof_path,
+            ctx.accounts.bubblegum_program.key(),
+            args.root,
+            data_hash,
+            args.creator_hash,
+            args.nonce,
+            args.index,
+            None, // signer passed through from ctx
+        )?;
+        pool.sellside_asset_amount = pool
+            .sellside_asset_amount
+            .checked_add(1)
+            .ok_or(MMMErrorCode::NumericOverflow)?;
+        sell_state.pool = pool.key();
+        sell_state.pool_owner = owner.key();
+        sell_state.asset_mint = asset_mint.key();
+        sell_state.cosigner_annotation = pool.cosigner_annotation;
+        sell_state.asset_amount = sell_state
+            .asset_amount
+            .checked_add(1)
+            .ok_or(MMMErrorCode::NumericOverflow)?;
+    } else {
+        transfer_compressed_nft(
+            &ctx.accounts.tree_authority.to_account_info(),
+            &ctx.accounts.payer.to_account_info(),
+            &ctx.accounts.payer.to_account_info(),
+            &ctx.accounts.owner.to_account_info(),
+            &ctx.accounts.merkle_tree,
+            &ctx.accounts.log_wrapper,
+            &ctx.accounts.compression_program,
+            &ctx.accounts.system_program, // Pass as Program<System> without calling to_account_info()
+            proof_path,
+            ctx.accounts.bubblegum_program.key(),
+            args.root,
+            data_hash,
+            args.creator_hash,
+            args.nonce,
+            args.index,
+            None, // signer passed through from ctx
+        )?;
+    }
+
+    // 4. Pool owner as buyer pay royalties to creators
+    let royalty_paid = pay_creator_fees_in_sol_cnft(
+        pool.buyside_creator_royalty_bp,
+        seller_receives,
+        &args.metadata_args,
+        creator_accounts,
+        buyside_sol_escrow_account.to_account_info(),
+        buyside_sol_escrow_account_seeds,
+        system_program.to_account_info(),
+    )?;
+
+    // 5. Seller pay buyer, prevent frontrun by pool config changes
+    let payment_amount = total_price
+        .checked_sub(lp_fee)
+        .ok_or(MMMErrorCode::NumericOverflow)?
+        .checked_sub(taker_fee as u64)
+        .ok_or(MMMErrorCode::NumericOverflow)?
+        .checked_sub(royalty_paid)
+        .ok_or(MMMErrorCode::NumericOverflow)?;
+    if payment_amount < args.min_payment_amount {
+        return Err(MMMErrorCode::InvalidRequestedPrice.into());
+    }
+
+    anchor_lang::solana_program::program::invoke_signed(
```

Comment:
seems like we can't change lamports for payer in this case since we don't own it?

```
Message: Transaction simulation failed: Error processing Instruction 0: instruction spent from the balance of an account it does not own. 
    Logs: 
    [
      "Program cmtDvXumGCrqC1Age74AVPhSRVXJMd8PJS91L8KbNCK consumed 41544 of 113844 compute units",
      "Program cmtDvXumGCrqC1Age74AVPhSRVXJMd8PJS91L8KbNCK success",
      "Program BGUMAp9Gq7iTEuizy4pqaxsTyUCBK68MDfK752saRPUY consumed 58978 of 130619 compute units",
      "Program BGUMAp9Gq7iTEuizy4pqaxsTyUCBK68MDfK752saRPUY success",
      "Program 11111111111111111111111111111111 invoke [2]",
      "Program 11111111111111111111111111111111 success",
      "Program 11111111111111111111111111111111 invoke [2]",
      "Program 11111111111111111111111111111111 success",
      "Program mmm3XBJg5gk8XJxEKBvdgptZz6SgK4tXvn36sodowMc consumed 139097 of 200000 compute units",
      "Program mmm3XBJg5gk8XJxEKBvdgptZz6SgK4tXvn36sodowMc failed: instruction spent from the balance of an account it does not own"
    ]. 
    Catch the `SendTransactionError` and call `getLogs()` on it for full details.
```

Reply:
you can't subtract lamports from the payer since it's not owned by the program. you can subtract it from the pda